packages/components/tests/e2e/specs/appmap/sequenceDiagram.spec.js

```diff
@@ -191,4 +191,25 @@ context('AppMap sequence diagram', () => {
       cy.get('.sequence-diagram').children('.lane').should('have.length', 8);
     });
   });
+
+  context('in initial compact view', () => {
+    beforeEach(() => {
+      cy.visit(
+        'http://localhost:6006/iframe.html?id=appland-diagrams-sequence--create-api-key&viewMode=story'
+      );
+    });
+
+    it("action is collapsed if it's 4+ level deep and has all function descendants", () => {
+      // Navigate to the first common ancestor after finding the span
+      // representing call label 'decode', to find the .collapse-expand
+      // div containing [+].
+      cy.contains('span', 'decode')
+        .first()
+        .parent()
+        .parent()
+        .parent()
+        .find('.collapse-expand')
+        .should('have.class', 'collapsed');
+    });
+  });
```

Comment:
Can you add tests to cover all possible combinations?

- If the action is 2+ levels deep, has only function descendants, but it is the selected action, then it shouldn't be collapsed.
- If the action is 2+ levels deep, is not the selected action, but has a sql query as a descendant, then it shouldn't be collapsed.
- etc